.github/workflows/reusable_tests_pure_python.yml

```diff
@@ -65,15 +65,17 @@ jobs:
       - name: Install dependencies
         run: |
           pip install pytest numpy
-          cd ${{inputs.python_folder}}
-          pip install .
       - name: Test with coverage pytest
         if: ${{inputs.test_code_coverage}}
         run: |
           pytest tests/ --cov=${{inputs.python_folder}} --cov-fail-under=80
+          cd ${{inputs.python_folder}}
+          pip install .
       - name: Test without coverage pytest
         if: ${{!inputs.test_code_coverage}}
         run: |
+          cd ${{inputs.python_folder}}
+          pip install .
           pytest tests/
```

Comment:
With this we might again run into the issue that in the import statements of the python tests the folder versions of the Python files will be imported and not the just installed package. 

We could instead cd into tests and run pytest inside that folder.